.github/copilot-instructions.md

```diff
@@ -0,0 +1,262 @@
+- Update both language versions simultaneously when making documentation changes
+- Ensure images and diagrams are accessible from both language versions
+
+#### Cross-Language Linking
+- Use named anchors to ensure stable URLs across languages
+- Test links in both languages after updates
+- Maintain parallel navigation structures in both `sidebars.js` and Japanese i18n
+
+#### Image and Media Guidelines
+- Store images in the English docs directory and reference from both languages
+- Use descriptive filenames that indicate the content
+- Provide appropriate alt text for accessibility
+- Optimize images for web display while maintaining clarity
+
+## Testing and Quality Assurance
+
+### Unit Testing
+- Write unit tests for all core business logic in `UnitTests~/`
+- Inherit from `TestBase` class for common test setup and teardown
+- Use Unity Test Runner with NUnit framework
+- Mock Unity objects appropriately in tests using `CreatePrimitive()` or prefabs
+- Test both editor and runtime behavior where applicable
+- Use conditional compilation (`#if MA_VRCSDK3_AVATARS`) for VRChat-specific tests
+- Clean up created GameObjects in teardown methods to prevent test pollution
+- Use the established pattern for loading test assets via GUID references
+
+### Documentation Testing
+- Build documentation locally before committing: `cd docs~ && yarn build`
+- Test both English and Japanese versions
+- Verify all links work in both languages
+- Check that named anchors resolve correctly
+
+### Integration Testing
+- Test components in Unity with real avatar setups
+- Verify non-destructive behavior (original assets unchanged)
+- Test build processes with NDMF integration
+- Validate VRChat SDK compatibility
+
+## Build and Development Workflow
+
+### Documentation Development
+```bash
+# Navigate to docs directory
+cd docs~
+
+# Install dependencies
+yarn install
+
+# Start development server
+yarn start
+
+# Build for production
+yarn build
+
+# Type checking
+yarn typecheck
+```
+
+### Unity Development
+- Use Unity 2022.3.22f1 (as specified in project)
+- Test with VRChat SDK3 Avatars
+- Verify NDMF integration works correctly
+- Test both play mode and edit mode functionality
+
+## Common Patterns and Anti-Patterns
+
+### ✅ Good Practices
+- Always check for null references before accessing Unity objects
+- Use appropriate Unity lifecycle methods (`Awake`, `Start`, `OnEnable`)
+- Implement proper cleanup in `OnDisable` and `OnDestroy`
+- Use `[SerializeField]` for editor-exposed fields instead of public fields
+- Validate component configuration in `OnValidate()`
+- Use descriptive commit messages following conventional commits
+
+### ❌ Anti-Patterns
+- Don't modify objects that don't belong to the component
+- Don't assume specific avatar structures without validation
+- Avoid hardcoded strings for user-facing text (use localization)
+- Don't create circular dependencies between components
+- Avoid blocking the main thread with expensive operations
```

Comment:
Unfortunately it is usually not possible to avoid blocking the main thread with unity. Remove this line.